api/controllers/profile-cotroller.js

```diff
@@ -1,25 +1,63 @@
 const { User } = require('../models/user-model');
+const bcrypt = require('bcryptjs');
+const uniqueValidator = require('mongoose-unique-validator');
+const mongoose = require('mongoose');
 
 const updateProfile = async (req, res) => {
     try {
-        const { id, user } = req.body;
-        const updateUserInfo = await User.findByIdAndUpdate(id, {$set: user });
-
-        return res.status(200).json(updateUserInfo);
+        const userId = req.userData.id;
+        const user = await User.findById(userId);
+        const userSchema = mongoose.Schema({
+            username: { type: String, unique: true },
+            email: { type: String, index: true, unique: true },
+        });
+        userSchema.plugin(uniqueValidator);
+        if (user.username === req.body.name) await User.findByIdAndUpdate(
```

Comment:
use **$set**  for updating multiple fields
https://stackoverflow.com/questions/37267042/mongoose-findoneandupdate-updating-multiple-fields